### PR TITLE
zh-Hans: Translate Pastebin solution

### PR DESCRIPTION
**Pull request by kevinxuv**
@sqrthree pls review.

### CODE REVIEW

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+为了删除过期的pastes，我们可以直接搜索**SQL数据库**中所有的过期时间比当前时间更早的记录，
+所有过期的记录将从这张表里面删除（或者将其标记为过期）
+
+## 第四步：扩展这个设计
+
+> 给定约束条件，识别和解决瓶颈。
+
+![Imgur](http://i.imgur.com/4edXG0T.png)
+
+**Important: 不要简单的从最初的设计直接跳到最终的设计**
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: `Important`=> `重要提示`

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+* [SQL write master-slave failover](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#故障切换)
+* [主从复制](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#主从复制)
+* [一致性模式](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#一致性模式)
+* [可用性模式](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#可用性模式)
+
+**分析存储数据库** 可以用比如Amazon Redshift或者Google BigQuery这样的数据仓库解决方案。
+
+一个像Amazon S3这样的**对象存储**，可以轻松处理每月12.7 GB的新内容约束。
+
+要处理每秒40 *平均*读请求(峰值更高)，其中热点内容的流量应该由**内存缓存**处理，而不是数据库。**内存缓存**对于处理分布不均匀的流量和流量峰值也很有用。只要副本没有陷入复制写的泥潭，**SQL Read Replicas**应该能够处理缓存丢失。
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: `要处理每秒40 *平均*读请求(峰值更高)` => `要处理每秒 **平均** 40 次的读请求(峰值更高)，`

**Comment by sqrthree** [comment]
In addition, I noticed that this PR contains a content [changes](https://github.com/donnemartin/system-design-primer/pull/273/files#diff-a9ff03ccee3fb02ddcfa48f18d1f86b6). Looks like it's a typo fix. Do we need to split it with a separate PR? @kevinxuv @donnemartin

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 3]
```diff
@@ -0,0 +1,331 @@
+# 设计 Pastebin.com (或者 Bit.ly)
+
+*Note: 为了避免重复，当前文档直接链接到[系统设计主题](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#系统设计主题的索引)的相关区域，请参考链接内容以获得综合的讨论点、权衡和替代方案。*
```
[the remark this answers is not in the document]
Comment (kevinxuv) [reply solutions/system_design/pastebin/README-zh-Hans.md]: 👍

**Comment by kevinxuv** [comment]
@sqrthree 
* Thanks for your review,  they are all good suggestions.
* For the typo fix in the origin english version,  it not a big fix, so I removed this change in this pr, maybe these typo will be fixed in next pr.

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        (2016-01, url0), 2
+        (2016-01, url1), 1
+        """
+        yield key, sum(values)
+```
+
+### 用例： 服务删除过期的 pastes
+
+为了删除过期的 pastes，我们可以直接搜索 **SQL 数据库** 中所有的过期时间比当前时间更早的记录，
+所有过期的记录将从这张表里面删除（或者将其标记为过期）
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: Punctuation is missing.

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+    digits = digits.reverse
+```
+
+* 取输出的前 7 个字符，结果会有 62^7 个可能的值，应该足以满足在 3 年内处理 3.6 亿个短链接的约束：
+
+```python
+url = base_encode(md5(ip_address+timestamp))[:URL_LENGTH]
+```
+
+我们将会用一个公开的[**REST 风格接口 **](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#表述性状态转移rest)：
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: image

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+  * 内部通信 - [RPC](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#远程过程调用协议rpc)
+* [服务发现](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#服务发现)
+
+### 安全
+
+参考[安全](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#安全).
+
+### 延迟数字
+
+见[每个程序员都应该知道的延迟数](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#每个程序员都应该知道的延迟数).
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: 需要使用中文的句号

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+### 通信
+
+* 讨论权衡:
+  * 跟客户端之间的外部通信 - [HTTP APIs following REST](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#表述性状态转移rest)
+  * 内部通信 - [RPC](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#远程过程调用协议rpc)
+* [服务发现](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#服务发现)
+
+### 安全
+
+参考[安全](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#安全).
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: 需要使用中文的句号

File: solutions/system_design/pastebin/README-zh-Hans.md [hunk 8]
```diff
@@ -0,0 +1,330 @@
+# 设计 Pastebin.com (或者 Bit.ly)
+
+*Note: 为了避免重复，当前文档直接链接到[系统设计主题](https://github.com/donnemartin/system-design-primer/blob/master/README-zh-Hans.md#系统设计主题的索引)的相关区域，请参考链接内容以获得综合的讨论点、权衡和替代方案。*
```
Comment (sqrthree) [inline solutions/system_design/pastebin/README-zh-Hans.md]: image
It seems that it has not been fixed

Comment (kevinxuv) [reply solutions/system_design/pastebin/README-zh-Hans.md]: 问题是？没有get到。

Comment (sqrthree) [reply solutions/system_design/pastebin/README-zh-Hans.md]: 在中文中加粗比倾斜的阅读体验更好。

Comment (kevinxuv) [reply solutions/system_design/pastebin/README-zh-Hans.md]: got it.

**Comment by kevinxuv** [comment]
@sqrthree pls help to review again.

**Comment by sqrthree** [comment]
@donnemartin I think it's ready to be merged.

**Comment by donnemartin** [comment]
@kevinxuv @sqrthree awesome, thank you!

**Comment by kevinxuv** [comment]
@donnemartin I should thank you for your repo, I learn a lot from it,  and also I want to thanks to @sqrthree, you teach me a lot about translation, thank you all.

**Pull request merged**